llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -482,6 +508,26 @@ class WAWEdges : public ScheduleDAGMutation {
   };
 };
 
+// Adds WAW edges for scheduling in the context of the Scheduler.
+// This class extends WAWEdges to apply WAW edges using a Scheduler if available
+// It overrides the apply method to retrieve the Scheduler from the DAG if a
+// BasicBlock is present, otherwise, it uses nullptr.
+class SchedWAWEdges : public WAWEdges {
+  void apply(ScheduleDAGInstrs *DAG) override {
```

Comment:
Do we run this more the pre-ra machinescheduler as well? If we do, I think casting to `AIEScheduleDAGMI` is invalid.

Reply:
It will run only in post ra and also while interblock dag construction. I won’t run in pre ra machine scheduler